From 64e2af67577224878f1dda1aba26d7b8cb1f77c0 Mon Sep 17 00:00:00 2001
From: Patryk Kups <pkups@anaconda.com>
Date: Tue, 9 Dec 2025 16:39:46 +0100
Subject: [PATCH 3/7] point to our libs / headers for windows in setup.py

---
 python/setup.py | 23 +++++++++++++++++------
 1 file changed, 17 insertions(+), 6 deletions(-)

diff --git a/python/setup.py b/python/setup.py
index 60e0811..867a924 100755
--- a/python/setup.py
+++ b/python/setup.py
@@ -163,7 +163,8 @@ class build_ext_win(_build_ext):
     # Must pre-install sentencepice into build directory.
     arch = get_win_arch()
 
-    if os.path.exists('..\\build_{}\\root\\lib'.format(arch)):
+    #if os.path.exists('..\\build_{}\\root\\lib'.format(arch)):
+    if False:
       cflags = [
           '/std:c++17',
           '/I' + os.path.normpath('..\\build_{}\\root\\include'.format(arch)),
@@ -173,16 +174,26 @@ class build_ext_win(_build_ext):
           '..\\build_{}\\root\\lib\\sentencepiece_train.lib'.format(arch),
       ]
       libs.extend(find_abseil_lib('..\\build_{}\\third_party'.format(arch)))
-    elif os.path.exists('..\\build\\root\\lib'):
+    #elif os.path.exists('..\\build\\root\\lib'):
+    if True:
       cflags = [
           '/std:c++17',
-          '/I' + os.path.normpath('..\\build\\root\\include'),
+          #'/I' + os.path.normpath('..\\build\\root\\include'),
+          '/MD',
+          '/I' + os.environ["LIBRARY_INC"]
       ] + get_protobuf_includes()
       libs = [
-          '..\\build\\root\\lib\\sentencepiece.lib',
-          '..\\build\\root\\lib\\sentencepiece_train.lib',
+          #'..\\build\\root\\lib\\sentencepiece.lib',
+          #'..\\build\\root\\lib\\sentencepiece_train.lib',
+          # equivalent of -L$PREFIX/lib -lsentencepiece -lsentencepiece_train -lprotobuf
+          os.environ["LIBRARY_LIB"] + f"\\{x}.lib"
+          # protobuf actually has the lib-prefix in the name also on windows;
+          # since libsentencepiece is static on windows, we also need _its_
+          # host dependencies for the link interface, i.e. also abseil
+          for x in ["sentencepiece", "sentencepiece_train", "libprotobuf", "abseil_dll"]
       ]
-      libs.extend(find_abseil_lib('..\\build\\third_party'))
+      #libs.extend(find_abseil_lib('..\\build\\third_party'))
+      libs.extend(find_abseil_lib(os.environ["LIBRARY_LIB"]))
     else:
       # build library locally with cmake and vc++.
       if arch == 'amd64':
-- 
2.43.0

